src/Environment.php

```diff
@@ -380,6 +386,42 @@ public function getInlineParserCharacterRegex(): string
         return $this->inlineParserCharacterRegex;
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function addEventListener(string $eventClass, callable $listener, int $priority = 0): ConfigurableEnvironmentInterface
+    {
+        $this->assertUninitialized('Failed to add event listener.');
+
+        if (!isset($this->listeners[$eventClass])) {
+            $this->listeners[$eventClass] = new PrioritizedList();
+        }
+
+        $this->listeners[$eventClass]->add($listener, $priority);
+
+        return $this;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function dispatch(AbstractEvent $event): AbstractEvent
+    {
+        $this->initializeExtensions();
+
+        $type = get_class($event);
+
+        foreach ($this->listeners[$type] ?? [] as $listener) {
+            if ($event->isPropagationStopped()) {
+                break;
+            }
+
+            call_user_func($listener, $event);
```

Comment:
Can we not just do `$listener($event);`? It should be faster to execute too.

Reply:
Good call!